### PR TITLE
[SEDONA-116] Add ST_YMax and ST_YMin function

### PR DESCRIPTION
**Pull request by naharon2210**
## Is this PR related to a JIRA ticket?

- Yes, the URL of the assoicated JIRA ticket is https://issues.apache.org/jira/projects/SEDONA/issues/SEDONA-116.

## What changes were proposed in this PR?
Added ST_YMax and ST_YMin functions

## How was this patch tested?
Added unit tests as well.

## Did this PR include necessary documentation updates?
Yes it includes docs changes as well.

### CODE REVIEW

**Comment by jiayuasu** [comment]
@naharon2210 Please read the contributor guide and test your individual test case locally: https://sedona.apache.org/community/develop/#run-unit-tests

GitHub Action take lots of computing resources in ASF.

**Comment by naharon2210** [comment]
> @naharon2210 Please read the contributor guide and test your individual test case locally: https://sedona.apache.org/community/develop/#run-unit-tests
> 
> GitHub Action take lots of computing resources in ASF.

@jiayuasu I tested the code locally and there were no build errors. But there seems to be some java/scala build errors. How do I resolve this?

File: docs/api/sql/Function.md [hunk 1]
```diff
@@ -12,6 +12,37 @@ SELECT ST_Distance(polygondf.countyshape, polygondf.countyshape)
 FROM polygondf
 ```
 
+## ST_YMin
+
+Introduction: Return the minimum Y coordinate of A
+
+Format: `ST_Y_Min (A:geometry)`
+
+Since: `v1.0.0`
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:21]: should be v1.2.1

File: docs/api/sql/Function.md [hunk 2]
```diff
@@ -12,6 +12,37 @@ SELECT ST_Distance(polygondf.countyshape, polygondf.countyshape)
...
+
+Output : 0
+
+## ST_YMax
+
+Introduction: Return the minimum Y coordinate of A
+
+Format: `ST_YMax (A:geometry)`
+
+Since: `v1.0.0`
```
Comment (jiayuasu) [inline docs/api/sql/Function.md:36]: v1.2.1

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by naharon2210** [comment]
Applied the requested changes and solved merge conflicts.

File: docs/api/flink/Function.md [hunk 3]
```diff
@@ -56,6 +56,33 @@ Spark SQL example:
 SELECT ST_Buffer(polygondf.countyshape, 1)
 FROM polygondf
 ```
+## ST_YMin
```
Comment (jiayuasu) [inline docs/api/flink/Function.md:59]: Please read the doc carefully. version is wrong. Example is not complete with input/output

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**